src/lib/utils.ts

```diff
@@ -34,3 +37,47 @@ export const loadScriptTag = async (srcValue: string): Promise<void> => {
 
   await loadedPromise
 }
+
+export const getUserVariables = (): {userId: string, courseId: string, instance: string} | null => {
+  const userId = '2'
+  const courseId = window.M?.cfg.courseId
+  const instance = window.location.host
+  if (userId === undefined || courseId === undefined || instance === undefined) {
+    return null
+  }
+  return {
+    userId,
+    courseId,
+    instance
+  }
+}
+
+function getVariantMappings(host: string, courseid: string): string {
```

Comment:
There's a little bit of inconsistency where you call it `host` here and `instance` in the object returned by `getUserVariables`. I personally like `host` here, so maybe we can modify the other ones just for consistency?